solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java

```diff
@@ -867,26 +866,25 @@ public String getTaskKey(ZkNodeProps message) {
   }
 
 
+  // -1 is not a possible batchSessionId so -1 will force initialization of lockSession
   private long sessionId = -1;
   private LockTree.Session lockSession;
 
   @Override
-  public Lock lockTask(ZkNodeProps message, OverseerTaskProcessor.TaskBatch taskBatch) {
-    if (lockSession == null || sessionId != taskBatch.getId()) {
+  public Lock lockTask(ZkNodeProps message, long batchSessionId) {
+    if (sessionId != batchSessionId) {
       //this is always called in the same thread.
       //Each batch is supposed to have a new taskBatch
       //So if taskBatch changes we must create a new Session
-      // also check if the running tasks are empty. If yes, clear lockTree
-      // this will ensure that locks are not 'leaked'
-      if(taskBatch.getRunningTasks() == 0) lockTree.clear();
```

Comment:
This is the only usage of clear in the code, can we remove that method completely? Is this safe to not call clear?

Reply:
I hope (and think) it is... A lock can leak if an executor thread dies in a place where it shouldn't be dying (just before the try with the lock released in the finally.
Clearing all locks is not a solution IMO. If we do end up with lock leaks we should address those in a more elegant way (fix the leak).